src/ray/common/id.h

```diff
@@ -78,78 +96,209 @@ class UniqueID : public BaseID<UniqueID> {
+
+ public:
+  /// Length of `ActorID` in bytes.
+  static constexpr size_t kLength = kUniqueBytesLength + JobID::kLength;
+
+  /// Size of `ActorID` in bytes.
+  ///
+  /// \return Void
+  static size_t Size() { return kLength; }
+
+  /// Generate an `ActorID` randomly.
+  ///
+  /// \param job_id The job id to which this actor belongs.
+  /// \return The random `ActorID`.
+  static ActorID FromRandom(const JobID &job_id);
+
+  static ActorID FromRandom() = delete;
+
+  /// Constructor of `ActorID`.
+  ActorID() : BaseID() {}
+
+  /// Get the job id to which this actor belongs.
+  ///
+  /// \return The job id to which this actor belongs.
+  JobID JobId() const;
+
+ private:
+  uint8_t id_[kLength];
 };
 
 class TaskID : public BaseID<TaskID> {
+ private:
+  static constexpr size_t kUniqueBytesLength = 6;
+
  public:
+  static constexpr size_t kLength = kUniqueBytesLength + ActorID::kLength;
+
   TaskID() : BaseID() {}
-  static size_t Size() { return kTaskIDSize; }
+
+  static size_t Size() { return kLength; }
+
   static TaskID ComputeDriverTaskId(const WorkerID &driver_id);
 
+  /// Generate TaskID randomly.
+  ///
+  /// Note that the ActorID of this task should be NIL.
+  ///
+  /// \return A random TaskID.
+  static TaskID FromRandom();
+
+  /// Generate TaskID from the given actor id.
+  ///
+  /// \param actor_id
+  /// \return
+  static TaskID FromRandom(const ActorID &actor_id);
+
+  /// Get the id of the actor to which this task belongs.
+  ///
+  /// \return The `ActorID` of the actor which creates this task.
+  ActorID ActorId() const;
+
  private:
-  uint8_t id_[kTaskIDSize];
+  uint8_t id_[kLength];
 };
 
 class ObjectID : public BaseID<ObjectID> {
+private:
+  static constexpr size_t kIndexBytesLength = sizeof(ObjectIDIndexType);
+
+  static constexpr size_t kFlagsBytesLength = sizeof(ObjectIDFlagsType);
+
  public:
+  /// The maximum number of objects that can be returned or put by a task.
+  static constexpr int64_t kMaxObjectIndex = ((int64_t) 1 << kObjectIdIndexSize) - 1;
+
+  /// The length of ObjectID in bytes.
+  static constexpr size_t kLength = kIndexBytesLength + kFlagsBytesLength + TaskID::kLength;
+
   ObjectID() : BaseID() {}
-  static size_t Size() { return kUniqueIDSize; }
+
+  /// The maximum index of object.
+  ///
+  /// It also means the number of objects(put or return) of one task.
+  ///
+  /// \return The maximum index of object.
+  static uint64_t MaxObjectIndex() { return kMaxObjectIndex; }
+
+  static size_t Size() { return kLength; }
+
+  /// Generate ObjectID by the given binary string of a plasma id.
+  ///
+  /// \param from The binary string of the given plasma id.
+  /// \return The ObjectID converted from a binary string of the plasma id.
+  static ObjectID FromPlasmaIdBinary(const std::string &from);
+
+  /// Generate an random actor cursor object id by the given actor.
+  ///
+  /// \param actor_id TODO(qwang)
+  /// \return
+  static ObjectID GenerateActorDummyObjectId(const ActorID &actor_id);
```

Comment:
Please complete this header to be more clear. I would also rename this to something like `ActorCreationDummyObjectId`.